tests/Unit/DataStructures/Test_VectorImplHelper.hpp

```diff
@@ -0,0 +1,69 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions for manipulating tuples
+
+#pragma once
+
+#include <tuple>
+#include <utility>
+
+#include "Utilities/Gsl.hpp"
+
+
+namespace TestHelpers {
+namespace VectorImpl {
+//{@
+/*!
+ * \brief Perform a function over combinatoric subsets of a std::tuple
+ *
+ * \details
+ * If given a function object, the apply_tuple_combinations function will
+ * apply the function to all ordered combinations. For instance, if the tuple
+ * contains elements (a,b,c) and the function f accepts two arguments it will be
+ * executed on all nine combinations of the three possible
+ * arguments. Importantly, the function f should then be valid for all such
+ * argument combinations.
+ *
+ * \typaram ArgLength `apply_tuple_combinations` requires an explicit template
```

Comment:
Not if I want to use it for lambdas (which I really do). From what I understand, lambdas can't be used in any of the clever const deductions that tt:function_info uses, partially because the `auto`s in lambdas don't yield well-determined function types.